flink-connector-prometheus/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusTimeSeriesConverter.java

```diff
@@ -22,42 +22,22 @@
 import org.apache.flink.connector.base.sink.writer.ElementConverter;
 import org.apache.flink.connector.prometheus.sink.prometheus.Types;
 
+import java.util.UnknownFormatConversionException;
+
 /**
  * Converts the sink input {@link PrometheusTimeSeries} into the Protobuf {@link Types.TimeSeries}
  * that are sent to Prometheus.
  */
 @PublicEvolving
-public class PrometheusTimeSeriesConverter
-        implements ElementConverter<PrometheusTimeSeries, Types.TimeSeries> {
-
-    private static final String METRIC_NAME_LABEL_NAME = "__name__";
+public class PrometheusTimeSeriesConverter<IN> implements ElementConverter<IN, Types.TimeSeries> {
 
     @Override
-    public Types.TimeSeries apply(PrometheusTimeSeries element, SinkWriter.Context context) {
-        Types.TimeSeries.Builder builder =
-                Types.TimeSeries.newBuilder()
-                        .addLabels(
-                                Types.Label.newBuilder()
-                                        .setName(METRIC_NAME_LABEL_NAME)
-                                        .setValue(element.getMetricName())
-                                        .build());
-
-        for (PrometheusTimeSeries.Label label : element.getLabels()) {
-            builder.addLabels(
-                    Types.Label.newBuilder()
-                            .setName(label.getName())
-                            .setValue(label.getValue())
-                            .build());
-        }
-
-        for (PrometheusTimeSeries.Sample sample : element.getSamples()) {
-            builder.addSamples(
-                    Types.Sample.newBuilder()
-                            .setValue(sample.getValue())
-                            .setTimestamp(sample.getTimestamp())
-                            .build());
+    public Types.TimeSeries apply(IN element, SinkWriter.Context context) {
```

Comment:
As my previous point: if we still support PrometheusTimeSeries only, is it worth breaking the public API?

Reply:
Answered in previous point: we will not introduce breaking change